notebooks/228-clip-zero-shot-image-classification/228-clip-zero-shot-image-classification.ipynb

```diff
@@ -0,0 +1,294 @@
+    "`CLIPProcessor` class is wrapper for input data preprocessing. It includes both encode the text using tokenizer and prepare the images."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from transformers import CLIPProcessor, CLIPModel\n",
+    "\n",
+    "# load pretrained model\n",
+    "model = CLIPModel.from_pretrained(\"openai/clip-vit-base-patch16\")\n",
+    "# load preprocessor for model input\n",
+    "processor = CLIPProcessor.from_pretrained(\"openai/clip-vit-base-patch16\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from typing import List\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "from PIL import Image\n",
+    "\n",
+    "\n",
+    "def visualize_result(image:Image, labels:List[str], probs:np.ndarray, top:int = 5):\n",
+    "    \"\"\" \n",
+    "    Utility function for visualization classification results\n",
+    "    params:\n",
+    "      image: input image\n",
+    "      labels: list of classification labels\n",
+    "      probs: model predicted softmaxed probabilities for each label\n",
+    "      top: number of the highest probability results for visualization\n",
+    "    returns:\n",
+    "      None\n",
+    "    \"\"\"\n",
+    "    plt.figure(figsize=(64, 64))\n",
+    "    top_labels = np.argsort(-probs)[:min(top, probs.shape[0])]\n",
+    "    top_probs = probs[top_labels]\n",
+    "    plt.subplot(8, 8, 1)\n",
+    "    plt.imshow(image)\n",
+    "    plt.axis(\"off\")\n",
+    "\n",
+    "    plt.subplot(8, 8, 2)\n",
+    "    y = np.arange(top_probs.shape[-1])\n",
+    "    plt.grid()\n",
+    "    plt.barh(y, top_probs)\n",
+    "    plt.gca().invert_yaxis()\n",
+    "    plt.gca().set_axisbelow(True)\n",
+    "    plt.yticks(y, [labels[index] for index in top_labels])\n",
+    "    plt.xlabel(\"probability\")   "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run PyTorch model inference\n",
+    "\n",
+    "For performing classification, we should define labels and load image in RGB format. For getting model wider text context and improve guidance, we extend labels description using template \"This is a photo of a\".\n",
+    "Both list of label descriptions and image should be passed through processor for obtaining dictionary with input data in model specific format. Model predicts image-text similarity score in raw logits format, which can be normalized to [0, 1] range using `softmax` function. Then, we select labels with highest similarity score as final result.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "image = Image.open('data/coco.jpg')\n",
```

Comment:
Hi, Is this image should be pre-downloaded on machine ?

Reply:
sorry, looks like I forgot too push changes with uploading image

Reply:
ok, thank you

Reply:
added